### ISSUE TITLE
Can i have two Prolog instances simultaneously?

### ISSUE THREAD

**Issue by veryrandomname** [yuce/pyswip#74]
From looking at the code it seems like Prolog is started when the library is imported.
Is that necessary? For me this library would only be useful if I could start/stop Prolog processes at will.

**Comment by skliarpawlo** [comment]
+1 also it seems like multiple instances of Prolog share the same prolog state, and I cannot find a way to even reset the state, like cancel all consult calls for example

**Comment by stuz5000** [comment]
This is a limitation of SWI prolog. The C interface supported multiple execution threads, but only a single (global) database.

You will not need to start/stop prolog process though (they only run while executing a query).

**Comment by skliarpawlo** [comment]
@stuz5000 thank you! I cannot find a way to change the database either. Do you happen to know if there is any possibility to do revert or replace for `consult(...)` the same way as it can be done with `retract_all(...)` for single facts.

**Comment by stuz5000** [comment]
Yes. You can place all of your rules and facts into a single module that is
independent of the rest of the database. You can them query for and abolish
the contents of the module. Modules themselves cannot be cleaned up.
There was a recent question about this in the SWI Discourse message group

**Comment by stuz5000** [comment]
https://swi-prolog.discourse.group/t/threaded-queries-rulebase-independence/1236/11

**Comment by femartip** [comment]
As this Issue is still not closed. For a solution to this problem I am using pywip_notebook as it creates an instance of Prolog in different modules. I think this may resolve your issues.
